File: src/main/kotlin/ir/parham/seasonJob/Commands/Default/SaveAll.java
package ir.parham.seasonJob.Commands.Default;

import Libs.API.ir.parham.SeasonJobsAPI.Commands;
import Libs.API.ir.parham.SeasonJobsAPI.Senders.Message;
import ir.parham.SeasonJobsAPI.Actions.Job;
import ir.parham.SeasonJobsAPI.Actions.Member;
import org.bukkit.Bukkit;
import org.bukkit.OfflinePlayer;
import org.bukkit.command.CommandSender;
import org.jetbrains.annotations.NotNull;

import java.util.List;
import java.util.UUID;

public class SaveAll implements Commands
{

    @Override
    public void runner(@NotNull CommandSender sender, @NotNull String[] args) {
        Message message = new Message();

        if (sender.hasPermission("seasonjobs.save-all"))
        {
            sender.sendMessage(message.get(Bukkit.getOfflinePlayer(UUID.randomUUID()), "save-all-process"));
            new Member().saveAll();
            new Job().saveAll();
            sender.sendMessage(message.get(Bukkit.getOfflinePlayer(UUID.randomUUID()), "save-all-finished"));
        }
        else
        {
            // deny perm
            sender.sendMessage(message.get(Bukkit.getOfflinePlayer(UUID.randomUUID()), "denyPerm"));
        }
    }

    @Override
    public @NotNull List<String> completer(@NotNull CommandSender sender, @NotNull String[] args) {
        return List.of();
    }

    @Override
    public void remover(@NotNull OfflinePlayer target) {

    }

    @Override
    public void adder(@NotNull OfflinePlayer target) {

    }

    @Override
    public void setter(@NotNull OfflinePlayer target) {

    }
}

File: src/main/kotlin/ir/parham/seasonJob/Commands/Default/SaveAll.kt
package ir.parham.seasonJob.Commands.Default

import Libs.API.ir.parham.SeasonJobsAPI.Commands
import Libs.API.ir.parham.SeasonJobsAPI.Senders.Message
import ir.parham.SeasonJobsAPI.Actions.Job
import ir.parham.SeasonJobsAPI.Actions.Member
import org.bukkit.Bukkit
import org.bukkit.OfflinePlayer
import org.bukkit.command.CommandSender
import java.util.*
import kotlin.collections.List

class SaveAll : Commands {
    override fun runner(sender: CommandSender, args: Array<out String>) {
        val message = Message()

        if (sender.hasPermission("seasonjobs.save-all")) {
            sender.sendMessage(message.get(Bukkit.getOfflinePlayer(UUID.randomUUID()), "save-all-process"))
            Member().saveAll()
            Job().saveAll()
            sender.sendMessage(message.get(Bukkit.getOfflinePlayer(UUID.randomUUID()), "save-all-finished"))
        } else {
            // deny perm
            sender.sendMessage(message.get(Bukkit.getOfflinePlayer(UUID.randomUUID()), "denyPerm"))
        }
    }

    override fun completer(sender: CommandSender, args: Array<out String>): List<String> {
        TODO("Not yet implemented")
    }

    override fun remover(target: OfflinePlayer) {
    }

    override fun adder(target: OfflinePlayer) {
    }

    override fun setter(target: OfflinePlayer) {
    }
}